backend/src/data/repositories/worker/worker.repository.ts

```diff
@@ -86,6 +90,24 @@ class Worker {
     return Worker.modelToEntity(worker, groupIds);
   }
 
+  async getGroupIdsByTenant(
+    groupIds: string[],
+    tenantId: string,
+  ): Promise<string[]> {
+    const groups = await this.#GroupModel
+      .query()
+      .select('id')
+      .where({ tenantId });
+
+    const groupIdsByTenant = groups
+      .map((group) => group.id)
+      .filter((id) => {
+        return groupIds.some((groupId) => groupId === id);
+      });
+
+    return groupIdsByTenant;
+  }
```

Comment:
What does this method should do? Why `groupsIds` param is passed here? Remember, repository is only for data aceess. If you need to postprocess/postfilter data It's better to move this logic outside of repository.